src/Workspaces/VisualBasic/Portable/ObsoleteSymbol/VisualBasicObsoleteSymbolService.vb

```diff
@@ -0,0 +1,53 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.ObsoleteSymbol
+Imports Microsoft.CodeAnalysis.PooledObjects
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ObsoleteSymbol
+    <ExportLanguageService(GetType(IObsoleteSymbolService), LanguageNames.VisualBasic)>
+    <[Shared]>
+    Friend Class VisualBasicObsoleteSymbolService
+        Inherits AbstractObsoleteSymbolService
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+            MyBase.New(SyntaxKind.DimKeyword)
+        End Sub
+
+        Protected Overrides Sub ProcessDimKeyword(result As ArrayBuilder(Of TextSpan), semanticModel As SemanticModel, token As SyntaxToken, cancellationToken As CancellationToken)
+            Dim localDeclaration = TryCast(token.Parent, LocalDeclarationStatementSyntax)
+            If localDeclaration Is Nothing Then
+                Return
+            End If
+
+            If localDeclaration.Declarators.Count <> 1 Then
```

Comment:
nit.  i don't see why we can't do this per declarator.  just striking out the identifier if it has an obsolete type seems like it would work.

Reply:
We could, but there are some cases I wasn't sure about:

1. If there are more than one declarator, are all declarators required to have the same type?
1. If there are more than one item in `Names`, are all locals declared by these names required  to have the same type?

Reply:
> If there are more than one declarator, are all declarators required to have the same type?

No.  This is legal:

`Dim x As String, y As Integer`

`If there are more than one item in Names, are all locals declared by these names required to have the same type?`

Yes.  All would have the same type.

Note: this is why i think it's better to just strikeout the *name* of the thing, vs the 'dim'.  

> Since the overwhelming majority of cases involve one declarator with one name, it seemed like there are diminishing returns to handling the other cases. Happy to add them though if it will be valuable.

I don't disagree.  You are certainly covering the 99.9% case.  But it seemed actually trivial to just walk the declarators, and walk the names, and then strike them out if obsolete.  

The major reason i want this, tbh, is that i don't like the special-casing of hte `dim` kind and all that jazz.  It seems simpler, and 100% encompassing to just hit the variables themselves and mark those.

--

Note: this is a NIT.  It was more curiosity and suggestion than anything else.  If you want to keep as is, no problem at all.

Reply:
➡️ Added `TestDeclarators` specifically for this